test_analyzer_streamline_single.py

```diff
@@ -211,6 +296,151 @@ def aper_ratio(x, next_x, prev_x):
         c6.SetRightMargin( 5.*c6.GetRightMargin() )
 	hists_aper[2].Draw("COLZ")
 
+#c7 = r.TCanvas("c7", "hist7 canvas", 600,  500)
+#hists_projx[0].SetFillColor(2)
+#hists_projx[0].SetLineColor(4)
+#hists_projx[0].SetLineStyle(2)
+#c7.SetRightMargin( 5.*c7.GetRightMargin() )
+#hists_projx[0].Draw()
+
+#c8 = r.TCanvas("c8", "hist8 canvas", 600,  500)
+#hists_projx[1].SetFillColor(2)
+#hists_projx[1].SetLineColor(4)
+#hists_projx[1].SetLineStyle(2)
+#c8.SetRightMargin( 5.*c8.GetRightMargin() )
+#hists_projx[1].Draw()
+
+#c9 = r.TCanvas("c9", "hist9 canvas", 600,  500)
+#hists_projx[2].SetFillColor(2)
+#hists_projx[2].SetLineColor(4)
+#hists_projx[2].SetLineStyle(2)
+#c9.SetRightMargin( 5.*c9.GetRightMargin() )
+#hists_projx[2].Draw()
+
+#c10 = r.TCanvas("c10", "hist10 canvas", 600,  500)
+#hists_projy[0].SetFillColor(2)
+#hists_projy[0].SetLineColor(4)
+#hists_projy[0].SetLineStyle(2)
+#c10.SetRightMargin( 5.*c10.GetRightMargin() )
+#hists_projy[0].Draw()
+
+#c11 = r.TCanvas("c11", "hist11 canvas", 600,  500)
+#hists_projy[1].SetFillColor(2)
+#hists_projy[1].SetLineColor(4)
+#hists_projy[1].SetLineStyle(2)
+#c11.SetRightMargin( 5.*c11.GetRightMargin() )
+#hists_projy[1].Draw()
+
+#c12 = r.TCanvas("c12", "hist12 canvas", 600,  500)
+#hists_projy[2].SetFillColor(2)
+#hists_projy[2].SetLineColor(4)
+#hists_projy[2].SetLineStyle(2)
+#c12.SetRightMargin( 5.*c12.GetRightMargin() )
+#hists_projy[2].Draw()
+
+
+	c13 = r.TCanvas("c13", "hist16 canvas", 600,  500)
+	hists_odds[0].SetFillColor(3)
+	hists_odds[0].SetLineColor(5)
+	hists_odds[0].SetLineStyle(2)
+	hists_odds_adj[0].SetFillColor(2)
+	hists_odds_adj[0].SetLineColor(4)
+	hists_odds_adj[0].SetLineStyle(2)
+	hists_odds_adj[0].Divide(hists_odds[0])
+	c13.SetRightMargin( 5.*c13.GetRightMargin() )
+	leg4 = r.TLegend(0.8,0.7,1.0,0.9)
+	leg4.AddEntry(hists_odds_adj[0],"Tagger","L")
+	hists_odds_adj[0].Draw("he")
+
+	#c14 = r.TCanvas("c14", "hist17 canvas", 600,  500)
+	hists_odds[1].SetFillColor(3)
+	hists_odds[1].SetLineColor(5)
+	hists_odds[1].SetLineStyle(2)
+	hists_odds_adj[1].SetFillColor(4)
+	hists_odds_adj[1].SetLineColor(4)
+	hists_odds_adj[1].SetLineStyle(2)
+	hists_odds_adj[1].Divide(hists_odds[1])
+	#c14.SetRightMargin( 5.*c14.GetRightMargin() )
+	leg4.AddEntry(hists_odds_adj[1],"Upstream","L")
+	hists_odds_adj[1].Draw("hesame")
+
+	#c15 = r.TCanvas("c15", "hist18 canvas", 600,  500)
+	hists_odds[2].SetFillColor(3)
+	hists_odds[2].SetLineColor(5)
+	hists_odds[2].SetLineStyle(2)
+	hists_odds_adj[2].SetFillColor(5)
+	hists_odds_adj[2].SetLineColor(4)
+	hists_odds_adj[2].SetLineStyle(2)
+	hists_odds_adj[2].Divide(hists_odds[2])
+	#c15.SetRightMargin( 5.*c15.GetRightMargin() )
+	leg4.AddEntry(hists_odds_adj[2],"Downstream","L")
+	leg4.Draw()
+	hists_odds_adj[2].Draw("hesame")
+
```

Comment:
see how you could do this in a loop, using the fact that you have an array of histograms? you could also use an iterator to automatically set a different line color for each histogram.